### ISSUE TITLE
Fix S2952 FN/FP: FNs on some Invocation kinds and FP on additional disposal

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6229]
Rule S2952 does not report on these kinds of invocations (from the sample of the rule):
```cs
  this?.fs?.Dispose(); // FN
  this.fs?.Dispose();  // FN
  fs?.Dispose();       // FN
  this!.fs?.Dispose(); // FN
```
Rule S2952 Does report in this case, while it shouldn't
```cs
  public void Cleanup() => fs.Dispose(); // FP: We should not report in this case as the disposal is taking place in Dispose() too
  public void Dispose() => fs.Dispose();
```
Originally reported by https://discuss.sonarsource.com/t/11777

**Comment by martin-strecker-sonarsource** [comment]
Peach validation revealed some FPs:
* Disposed in `virtual Dispose(bool disposing)` of the [dispose pattern](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern)
* Diposed in [DisposeAsync](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync)
* Disposal delegated in Dispose to another method.
